ipapython/ipaldap.py

```diff
@@ -1749,3 +1752,239 @@ def get_ldap_uri(host='', port=389, cacert=None, ldapi=False, realm=None,
             return 'ldap://%s' % format_netloc(host, port)
         else:
             raise ValueError('Protocol %r not supported' % protocol)
+
+
+class CacheEntry:
+    def __init__(self, entry=None, attrs_list=None, exception=None,
+                 get_effective_rights=False, all=False):
+        self.entry = entry
+        self.attrs_list = attrs_list
+        self.exception = exception
+        self.all = all
+
+
+class LDAPCache(LDAPClient):
+    """A very basic LRU Cache using an OrderedDict"""
+
+    def __init__(self, ldap_uri, start_tls=False, force_schema_updates=False,
+                 no_schema=False, decode_attrs=True, cacert=None,
+                 sasl_nocanon=True, enable_cache=True, cache_size=100):
+
+        self.cache = OrderedDict()
+        self._cache_enabled = enable_cache
```

Comment:
What do you think about setting `self._cache_enabled`to `enable_cache and cache_size > 0`, as setting `cache_size` to `0` effectively disables cache?

Reply:
Good idea, done.